### DISCUSSION TITLE
Is there a way to get all current touch locations from a pointer event?

### DISCUSSION THREAD

**Discussion by pauldendulk** [AvaloniaUI/Avalonia#14723] [Q&A] [answered]
In Mapsui we have our own way to detect pinch and drag from the full set of touch locations. On some platforms we can request those from within the touch events. In Avalonia we keep track of the full set of touch locations locations. Our code could be simplified if we could request the full set of touch locations from Avalonia.

**Comment by maxkatz6** [comment, marked as answer]
No, Avalonia doesn't keep track of touch locations. Internally, we have a set of current touch pointers, but without keeping their location.

**Reply by pauldendulk** [reply]
What do you mean with "we have a set of current touch pointers"? As far as I can see the pointer event is always about just one location.

**Reply by thevortexcloud** [reply]
I am guessing that is referring to multi touch scenarios (as there are [multi touch gestures](https://docs.avaloniaui.net/docs/reference/gestures/pinchgesturerecognizer)). But I can't say for sure.

**Reply by maxkatz6** [reply]
> As far as I can see the pointer event is always about just one location.

PointerEvents are about one location. Correct.

But Avalonia keep tracks of all touches independently, guarantying that `args.Pointer` object is consistently the same across all events for the single touch (from press to release). https://github.com/AvaloniaUI/Avalonia/blob/master/src/Avalonia.Base/Input/TouchDevice.cs#L24

But again, there is no API to get all current pointers.

**Reply by pauldendulk** [reply]
Clear, thanks. For us this is not blocking. The Avalonia team might consider exposing the raw touch locations (x, y and id) as additional data in the `PointerEventArgs`.
